packages/web/src/ui/views/viewer.js

```diff
@@ -63,39 +63,40 @@ const viewer = (state, i18n) => {
     render = prepareRender(viewerOptions)
     const gestures = require('most-gestures').pointerGestures(el)
 
+    let delta = [0,0];
+    let deltaPan = [0,0];
+    let timer;
+
     // rotate
     gestures.drags
       .forEach((data) => {
         const ev = data.originalEvents[0]
-        const shiftKey = (ev.shiftKey === true) || (ev.touches && ev.touches.length > 2)
-        if (!shiftKey) {
-          const now = Date.now()
-          const ms = now - lastRotate
-          if (ms > (1000 / rotateRate)) {
-            const delta = [data.delta.x, data.delta.y].map((d) => -d)
-            const updated = orbitControls.rotate({ controls, camera, speed: rotateSpeed }, delta)
-            controls = { ...controls, ...updated.controls }
-            lastRotate = now
-          }
-        }
-      })
-    // pan
-    gestures.drags
-      .forEach((data) => {
-        const ev = data.originalEvents[0]
+        const {x,y} = data.delta;
         const shiftKey = (ev.shiftKey === true) || (ev.touches && ev.touches.length > 2)
         if (shiftKey) {
-          const now = Date.now()
-          const ms = now - lastPan
-          if (ms > (1000 / panRate)) {
-            const delta = [data.delta.x, data.delta.y].map((d) => d)
-            const updated = orbitControls.pan({ controls, camera, speed: panSpeed }, delta)
-            camera.position = updated.camera.position
-            camera.target = updated.camera.target
-            lastPan = now
-          }
+          deltaPan[0] += x;
+          deltaPan[1] += y;          
+        }else{
+          delta[0] -= x;
+          delta[1] -= y;
         }
-      })
+        if(!timer) timer = requestAnimationFrame(doRotatePan);
```

Comment:
Having said the above, I can understand that the refresh rate of the rendering is fixed, in the same way using animation frame. So, it may make sense to limit the delta (camera) changes to the same rate.

Does this make sense?

Reply:
This code does not skip events (just skips calling rotate/pan for burst of events) so in that regards is better in following the mouse :) .

The reason to use requestAnimationFrame is to let browser decide the rate.
Browser does have own rate if CPU is not very busy, but can also reduce framerate further when under heavy load.

I did try it on my Android phone and seems to be working well, even better than jscad.xyz
also on my laptop touchpad, very smooooth :)

Reply:
Good to know that testing is with other devices. 👍 

OK. So, can the two loops be combined? I don't see why two loops would be required.

Also, have you tried removing the 'throttles' in updateAndRender? I'm wondering are necessary now.

Reply:
I have already combined drag and pan loops since they source same events
```
    gestures.drags
      .forEach((data) => {
```
I can remove throttles in updateAndRender but need to tweak code a bit so zoom/pan does not cause it to be called twice in the same animationFrame while rotation is happening.

just showing a grid on mobile is already hard enough for my phone, an in that case new code is no worse than old.

Also further consideration would be making updateAndRender be called only when needed instead on a timer.

I will also change zoom to use this principle (zoom to fit does not need this).